server/game/GameActions/DuelAction.ts

```diff
@@ -102,9 +103,17 @@ export class DuelAction extends CardGameAction {
             context.game.addMessage('The duel cannot proceed as at least one participant for each side has to be in play');
             return;
         }
+        var duel = new Duel(context.game, properties.challenger, cards, properties.type, properties.statistic);
+        if(properties.duringDuelAction) {
+            properties.duringDuelAction.duration = Durations.UntilEndOfDuel;
```

Comment:
Game actions are stateless now, so you cannot change the duration like this (I don't think).  You would have to change either the properties or the propertyFactory property of the game action, and that get's very messy.  I don't think there's a way to make it work without introducing state to game actions, which I don't want to do (having just done so much work to remove it)